src/endpoints/azurePipelines.ts

```diff
@@ -0,0 +1,88 @@
+import { getRecord } from '@helpers/getRecord';
+import { savePipelineToRecord } from '@helpers/savePipelineToRecord';
+
+/**
+ * Webhook Listener
+ *
+ */
+aha.on('azurePipelinesHook', async ({ headers, payload }) => {
+  const event = payload.eventType;
+
+  switch (event) {
+    case 'build.complete':
+      await handleBuildCompleted(payload);
+      break;
+    default:
+      break;
+  }
+});
+
+/**
+ * Handle Build Complete Event
+ *
+ * @param payload
+ * @returns
+ */
+const handleBuildCompleted = async (payload: Webhook.Payload) => {
+  const recordField = parsePayloadToPipeline(payload);
+  if (!recordField) {
+    return;
+  }
+
+  const pipeline = Object.values(recordField?.pipelines ?? {})?.[0];
+
+  // Make sure the MR is linked to its record.
+  let record = await getRecord(pipeline?.branch?.replace('refs/heads/', ''));
+  if (!record) {
+    record = await getRecord(pipeline?.commitMsg);
+  }
+
+  if (!record) {
+    return;
+  }
+
+  await savePipelineToRecord(record, recordField);
+};
+
+/**
+ * Parse webhook payload to extension field by Resource Version
+ *
+ * @param payload
+ * @returns
+ */
+const parsePayloadToPipeline = (payload: Webhook.Payload): IExtensionFieldPipeline => {
+  let pipeline: IExtensionFieldPipeline;
+  switch (payload.resourceVersion) {
+    case '1.0': {
```

Comment:
@sorza-jimmy Can you explain what's going on here? All of the build hooks I'm receiving from ADO have a resource version of 1.0, which this clause seems to reject.

Reply:
@trydionel  Yes. In resource version 1.0, the webhook payload is missing some infos like repository info, project info, and commit message.

Reply:
How did you get it to generate anything else? I haven't seen any other resource version.

Reply:
You can choose latest version or 2.0-preview.2 in webhook configuration
![image](https://user-images.githubusercontent.com/79676156/157526208-42613a69-4c86-4d63-8ca6-468156d3ceb4.png)
.

Reply:
Currently working in 2.0-preview.2 only.

Reply:
Ahhhhh perfect! That did the trick. I'll start reviewing in earnest now :)

Reply:
I suspect this is going to be a common failure point for users. Could you add some logging here that will help end users triage the issue? This will show up in the extension webhook invocation logs (I think – please confirm).

```suggestion
    case '1.0': {
      console.log("This extension requires that responses use resource version '2.0-preview.2'. Please update your service hook configuration in Azure DevOps.");
```